### ISSUE TITLE
Cache problems

### ISSUE THREAD

**Issue by RoobieBoobieee** [spatie/laravel-translation-loader#44]
Is there any way to clear the cache? 
I cleared all translations in the database to empty strings but when displaying them they still show the old values. Executing the command php artisan cache:clear does not do anything, neither does updating a translation via:
```
$languageLine = LanguageLine::find($request->id);
        $languageLine->text = [
            'en' => $request->en,
            'nl' => $request->nl,
            'fr' => $request->fr,
        ];
$languageLine->save();
```
The documentation says an update should invalidate the cache, does this mean it invalidates all cached line or just that language line?

TL;DR Cleared database, language lines still showing, don't know how to clear cache.

**Comment by introwit** [comment]
@RoobieBoobieee Hi :) If you have cleared the DB & ran cache clear as you said, it should do the required job. Can you try `php artisan view:clear` as an extra step?

**Comment by RoobieBoobieee** [comment]
That didn't work. 
Steps to reproduce:
* I set the translation to 'test" for group `team` and key `set_pin`
* 'test' shows in front-end/database 
* I ran my migration/seeding again `php artisan migrate:refresh --seed`.
* There is only one database field with group `team` and key `set_pin`.
* The field contains `{"en":"Enter a pin code",...}` as I set it in my seeder.
* The front end still shows 'test' even after `php artisan cache:clear` and `php artisan view:clear`.

Editing the translation via php updates it in the database as well as on the front end.

Truncating the table removes all translations from the front end. 
I ran `php artisan cache:clear` and `php artisan view:clear` after truncating and after seeding/migrating and it still shows 'test'

**Comment by RoobieBoobieee** [comment]
removing everything in `storage/framework/cache` does the job.

**Comment by RoobieBoobieee** [comment]
Looks like it is an issue where the created cache is not writable.
Which makes that the `php artisan cache:clear` command doesn't do anything.
It just fails without an error. 

Sorry for bothering you guys :)

**Comment by introwit** [comment]
No problem, Glad you found the solution Rob :)

**Comment by bilal186** [comment]
I have fix the issue by extending the LanguageLine model class in translation laoder

php artisan vendor:publish --provider="Spatie\TranslationLoader\TranslationServiceProvider" --tag="config"

change the model with your extended model and remove Cache::rememberForever by overriding getTranslationsForGroup method

'model' => LanguageLineExtended::class,

<?php

namespace App;

use Illuminate\Support\Facades\Log;
use Illuminate\Support\Arr;
use Illuminate\Support\Facades\Cache;
use Spatie\TranslationLoader\LanguageLine;

class LanguageLineExtended extends LanguageLine
{
    protected $table = 'language_lines';

    public static function getTranslationsForGroup(string $locale, string $group): array
    {

        return Cache::remember(static::getCacheKey($group, $locale), '15', function () use ($group, $locale) {
            return static::query()
                    ->where('group', $group)
                    ->get()
                    ->reduce(function ($lines, self $languageLine) use ($group, $locale) {
                        $translation = $languageLine->getTranslation($locale);

                        if ($translation !== null && $group === '*') {
                            // Make a flat array when returning json translations
                            $lines[$languageLine->key] = $translation;
                        } elseif ($translation !== null && $group !== '*') {
                            // Make a nesetd array when returning normal translations
                            Arr::set($lines, $languageLine->key, $translation);
                        }

                        return $lines;
                    }) ?? [];
        });
    }
}
